src/components/tutorial.jsx

```diff
@@ -94,34 +94,36 @@ Tutorial.checkIfCompleted = (tutorial, user, preferences) => {
     if (preferences && preferences.preferences && preferences.preferences.tutorials_completed_at && preferences.preferences.tutorials_completed_at[tutorial.id]) {
       return Promise.resolve(preferences.preferences.tutorials_completed_at[tutorial.id]);
     }
+    return Promise.resolve(false);
   } else {
     if (completedThisSession[tutorial.id]) {
       return Promise.resolve(completedThisSession[tutorial.id]);
     }
+    return Promise.resolve(false);
```

Comment:
I think lines 99-102 could be tidied up as just: `Promise.resolve(!!completedThisSession[tutorial.id])` which will return true or false depending on the presence of that id in the object. The above lines did have to have the existential checks first because it is a deeply nested object. This isn't. Also as noted in the above comment, we should be consistent with the type that is returned and here, it should be a boolean.